File: 程式語言 V2答案檔/Go/408.go
package main
import (
	"fmt"
)

func reverse(str string) string {
	result := ""
	for i := len(str) - 1; i >= 0; i--{
		result += string(str[i])
	}
	return result
}

func main(){
	var a string
	var b string
	fmt.Scanln(&a)
	fmt.Scanln(&b)
	if len(a) < 3 || len(a) > 20 || len(b) < 3 || len(b) > 20 {
		fmt.Println("error")
		return
	}
	
	fmt.Println(len(a))
	fmt.Println(len(b))
	fmt.Println(reverse(b) + reverse(a))
}

File: Programming Language(ver.2)/C#/408.cs
using System;

public class TQC408
{
    public static void Main(string[] args)
    {
        string inputStr1;
        string inputStr2;
        inputStr1 = Console.ReadLine();
        inputStr2 = Console.ReadLine();

        if (ValidInputLength(inputStr1) && ValidInputLength(inputStr2)) {
            Console.WriteLine(inputStr1.Length);
            Console.WriteLine(inputStr2.Length);
            Console.WriteLine(ReverseString(inputStr2) + ReverseString(inputStr1));
        }
        else {
            Console.WriteLine("error");
        }
    }

    public static bool ValidInputLength(string str) {
        return str.Length > 3 && str.Length <= 20;
    }

    public static string ReverseString(string str) {
        string result = "";
        for (int i = str.Length - 1; i >= 0; i--) {
            result += str[i];
        }
        return result;
    }
}